DSharpPlus/Clients/DiscordClient.cs

```diff
@@ -580,6 +581,116 @@ public async Task<DiscordUser> UpdateCurrentUserAsync(string username = null, Op
+        /// <param name="commands">The list of commands to overwrite with.</param>
+        /// <returns>The list of guild commands.</returns>
+        public Task<IReadOnlyList<DiscordApplicationCommand>> BulkOverwriteGuildApplicationCommandsAsync(ulong guildId, IEnumerable<DiscordApplicationCommand> commands) =>
+            this.ApiClient.BulkOverwriteGuildApplicationCommandsAsync(this.CurrentApplication.Id, guildId, commands);
+
+        /// <summary>
+        /// Creates or overwrites a guild slash command.
+        /// </summary>
+        /// <param name="guildId">The id of the guild to create the slash command in.</param>
+        /// <param name="command">The command to create.</param>
+        /// <returns>The created command.</returns>
+        public Task<DiscordApplicationCommand> CreateGuildApplicationCommandAsync(ulong guildId, DiscordApplicationCommand command) =>
+            this.ApiClient.CreateGuildApplicationCommandAsync(this.CurrentApplication.Id, guildId, command);
+
+        /// <summary>
+        /// Gets a slash command in a guild by its id.
+        /// </summary>
+        /// <param name="guildId">The id of the guild the slash command is in.</param>
+        /// <param name="commandId">The id of the command to get.</param>
+        /// <returns>The command with the id.</returns>
+        public Task<DiscordApplicationCommand> GetGuildApplicationCommandAsync(ulong guildId, ulong commandId) =>
+             this.ApiClient.GetGuildApplicationCommandAsync(this.CurrentApplication.Id, guildId, commandId);
+
+        /// <summary>
+        /// Edits a slash command in a guild.
+        /// </summary>
+        /// <param name="guildId">The id of the guild the slash command is in.</param>
+        /// <param name="commandId">The id of the command to edit.</param>
+        /// <param name="action">Action to perform.</param>
+        /// <returns>The edited command.</returns>
+        public async Task<DiscordApplicationCommand> EditGuildApplicationCommandAsync(ulong guildId, ulong commandId, Action<ApplicationCommandEditModel> action)
+        {
+            var mdl = new ApplicationCommandEditModel();
+            action(mdl);
+            var applicationId = this.CurrentApplication?.Id ?? (await this.GetCurrentApplicationAsync()).Id;
+            return await this.ApiClient.EditGuildApplicationCommandAsync(applicationId, guildId, commandId, mdl.Name, mdl.Description, mdl.Options);
+        }
+
+        /// <summary>
+        /// Deletes a slash command in a guild.
+        /// </summary>
+        /// <param name="guildId">The id of the guild to delete the slash command in.</param>
+        /// <param name="commandId">The id of the command.</param>
+        public Task DeleteGuildApplicationCommandAsync(ulong guildId, ulong commandId) =>
+            this.ApiClient.DeleteGuildApplicationCommandAsync(this.CurrentApplication.Id, guildId, commandId);
         #endregion
```

Comment:
I don't know if we should have these in the regular client class considering these are already implemented for their respective objects.

Reply:
My reasoning was that a lot of times, people might not have the guild object and have just the id (for example private/single server bots which don't want global commands). I can change this, but I kept them in for convenience, though this does bloat the client with tons of methods.